### PR TITLE
Add resident type and booking approval

### PR DESCRIPTION
**Pull request by djensenius**

### CODE REVIEW

File: app/controllers/elevator_bookings_controller.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def create
     render json: { error: "invalid_token" }, status: :unauthorized and return false unless @user
 
     @elevator_booking = ElevatorBooking.new(elevator_booking_params)
+    @elevator_booking.approved = 0
```
Comment (MaggieShan) [inline app/controllers/elevator_bookings_controller.rb:25]: Just double checking what this is supposed to mean? 
Are all elevator bookings automatically not approved? (It seems that in ruby, all values including zero are evaluated to true).

File: db/schema.rb [hunk 2]
```diff
@@ -90,17 +91,8 @@
...
-    t.integer "unit"
-    t.string "email"
-    t.string "phone"
-    t.boolean "active"
-    t.boolean "admin"
-    t.datetime "created_at", precision: 6, null: false
-    t.datetime "updated_at", precision: 6, null: false
-    t.boolean "parking_admin"
-  end
+# Could not dump table "users" because of following StandardError
```
Comment (MaggieShan) [inline db/schema.rb:94]: This seems like an error with an invalid column type?

File: app/controllers/elevator_bookings_controller.rb [hunk 3]
```diff
@@ -50,17 +51,10 @@ def set_elevator_booking
...
-        unit: b.unit,
-        ownerType: b.ownerType,
-        name1: b.name1,
-        name2: b.name2,
-        phoneDay: b.phone_day,
-        phoneNight: b.phone_night,
-        deposit: b.deposit,
-        moveType: b.moveType
+        id: b.id, endTime: b.end, startTime: b.start, unit: b.unit,
+        ownerType: b.ownerType, name1: b.name1, name2: b.name2,
```
Comment (MaggieShan) [inline app/controllers/elevator_bookings_controller.rb:55]: Should this be user type instead of owner type?

File: db/schema.rb (hunk 2 above)
Comment (MaggieShan) [reply db/schema.rb:94]: Might it have something to do with the difference in owner and user type?

Comment (djensenius) [reply db/schema.rb:94]: This file should be deleted as we switched to `structure.db`

**Review by MaggieShan — APPROVED** [review]
lgtm

**Pull request merged**